--- a/feeds/packages/lang/rust/Makefile
+++ b/feeds/packages/lang/rust/Makefile
@@ -72,11 +72,12 @@
 endef
 
 define Host/Compile
-	$(RUST_SCCACHE_VARS) \
+    $(RUST_SCCACHE_VARS) \
 	CARGO_HOME=$(CARGO_HOME) \
 	TARGET_CFLAGS="$(TARGET_CFLAGS)" \
 	$(PYTHON) $(HOST_BUILD_DIR)/x.py \
 		--build-dir $(HOST_BUILD_DIR)/build \
+		--ci false \
 		dist build-manifest rustc rust-std cargo llvm-tools rust-src
 endef
